packages/flutter/lib/src/material/tab_bar_theme.dart

```diff
@@ -133,6 +151,42 @@ class TabBarTheme with Diagnosticable {
         && other.labelPadding == labelPadding
         && other.labelStyle == labelStyle
         && other.unselectedLabelColor == unselectedLabelColor
-        && other.unselectedLabelStyle == unselectedLabelStyle;
+        && other.unselectedLabelStyle == unselectedLabelStyle
+        && other.overlayColor == overlayColor
+        && other.splashFactory == splashFactory;
+  }
+}
+
+
+@immutable
+class _LerpColors implements MaterialStateProperty<Color?> {
```

Comment:
Interesting. I wonder if this might make a good candidate for a static function on `MaterialStateProperty` that could be used with any type.